### PR TITLE
Reorganize and update documentation: use Sphinx, readthedocs, add CMakeModules submodule

### PR DESCRIPTION
**Pull request by climbfuji**
1. Add documentation using restructured text with HTML and TeX/LaTeX (PDF) output.
2. Delete whatever is now in the sphinx documentation from `README.md`.
3. Move modulefile templates into common directory `modulefile_templates` under `doc`, add template for `git-lfs`
4. Add new submodule https://github.com/noaa-emc/CMakeModules as `doc/CMakeModules` to find `sphinx`.

The output is currently generated online from my user (@climbfuji) using `readthedocs` and is available here:

https://spack-stack.readthedocs.io/en/latest/
https://spack-stack.readthedocs.io/_/downloads/en/latest/pdf/

I also repeated the steps in section "Generating documentation" locally on my Mac to produce HTML and PDF output. Note that the PDF already contains the links to version `v0.0.1` instead of `latest`, not sure if this is what we want.

Once this PR is merged, we should set up the automatic integration at either JCSDA or NOAA-EMC. I am happy to take care of this at JCSDA.

### CODE REVIEW

**Comment by kgerheiser** [comment]
Are there any parts of these docs I can work on? I can take a section or something.

**Comment by climbfuji** [comment]
I am almost done :-) But thanks for asking. Will ping you later today xwhen it’s ready to review.

**Comment by climbfuji** [comment]
This is what I have at the moment, expect to finalize today: https://spack-stack.readthedocs.io/en/latest/

**Comment by climbfuji** [comment]
@edwardhartnett @srherbener I added you as reviewer so that you are aware of this effort. You don't need to review if you don't have time this week (in which case please remove yourself from the list of reviewers), but you are of course welcome to do so.

**Comment by climbfuji** [comment]
@kgerheiser I already made the changes for the spack v0.18.0 merge (essentially only removing `--reuse`).

File: doc/source/Quickstart.rst [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+The following instructions install a new spack environment on a pre-configured site (see :numref:`Section %s <Platforms_Preconfigured_Sites>` for a list of pre-configured sites and site-specific notes). Instructions for creating a new site config on an configurable system (i.e. a generic Linux or macOS system) can be found in :numref:`Section %s <Platform_New_Site_Configs>`.
+
+.. code-block:: console
+
+   # See a list of sites and apps
+   spack stack create env -h
+
+   # Create a pre-configured Spack environment in envs/<app>.<site>
+   # (copies site-specific, application-specific, and common config files into the environment directory)
+   spack stack create env --site hera --app jedi-fv3 --name jedi-fv3.hera
```
Comment (kgerheiser) [inline doc/source/Quickstart.rst]: `--specs jedi-fv3-env`

File: doc/source/Quickstart.rst [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+Create container
+------------------------
+
+.. code-block:: console
+
+   # See a list of preconfigured containers
+   spack stack create container -h
+
+   # Create container spack definition (spack.yaml) in directory envs/<spec>.<config>
+   spack stack create container docker-ubuntu-gcc-openmpi --app ufs-weather-model
```
Comment (kgerheiser) [inline doc/source/Quickstart.rst]: `--specs ufs-weather-model-env`

**Comment by kgerheiser** [comment]
I think this looks good. It's something we can iterate on.

**Review by kgerheiser — APPROVED** [review]

File: doc/source/Quickstart.rst (hunk 2 above)
Comment (climbfuji) [reply doc/source/Quickstart.rst]: Fixed, thanks!

File: doc/source/Quickstart.rst (hunk 1 above)
Comment (climbfuji) [reply doc/source/Quickstart.rst]: Fixed, thanks!

**Comment by kgerheiser** [comment]
I fixed the required checks

**Pull request merged**